### PR TITLE
UIREQ-459: Request staff notes: View note details

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
* Add `View staff note details` page

## Issue

[UIREQ-459](https://issues.folio.org/browse/UIREQ-459)

### CODE REVIEW

File: src/routes/NoteViewRoute.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { Redirect } from 'react-router';
+
+import { NoteViewPage } from '@folio/stripes/smart-components';
+
+import { formatNoteReferrerEntityData } from '../utils';
+import {
+  APP_ICON_NAME,
+} from '../constants';
```
Comment (Vladyslav-Velytskyi) [inline src/routes/NoteViewRoute.js:10]:
```suggestion
import { APP_ICON_NAME } from '../constants';
```

File: test/bigtest/network/factories/note.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { Factory, faker } from '@bigtest/mirage';
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/network/factories/note.js:1]:
```suggestion
import {
  Factory,
  faker,
} from '@bigtest/mirage';
```

File: test/bigtest/tests/view-request-test.js [hunk 3]
```diff
@@ -96,10 +98,29 @@ describe('View request page', () => {
         await viewRequest.staffNotesAccordion.clickNewButton();
       });
 
-      it('should display new staff notes page', function () {
+      it('should displayed new staff notes page', function () {
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/tests/view-request-test.js:101]: Please, return changes https://github.com/folio-org/ui-requests/pull/590#discussion_r444056703

File: test/bigtest/interactors/view-request.js [hunk 4]
```diff
@@ -35,6 +35,10 @@ import { contains } from './helpers';
   staffNotesAccordion = new NotesAccordion('#staff-notes');
 
   itemAccordionClick = clickable('#accordion-toggle-button-item-info');
+
+  whenNotesLoaded() {
+    return this.when(() => this.staffNotesAccordion && this.staffNotesAccordion.notes().length > 0);
```
Comment (VictorSoroka1) [inline test/bigtest/interactors/view-request.js:40]: I would suggest using `isEmpty` from `lodash` instead.
```suggestion
    return this.when(() => this.staffNotesAccordion && !isEmpty(this.staffNotesAccordion.notes()));
```

File: src/routes/NoteViewRoute.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}) => {
+  const { noteId } = match.params;
+
+  return location.state
+    ? (
+      <NoteViewPage
+        entityTypeTranslationKeys={{ request: 'ui-requests.notes.entityType.request' }}
+        entityTypePluralizedTranslationKeys={{ request: 'ui-requests.notes.entityType.request.pluralized' }}
+        navigateBack={history.goBack}
+        onEdit={() => {}}
```
Comment (VictorSoroka1) [inline src/routes/NoteViewRoute.js:23]: I would suggest using `noop` from `lodash`.

```suggestion
        onEdit={noop}
```

File: src/routes/NoteViewRoute.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        paneHeaderAppIcon={APP_ICON_NAME}
+        referredEntityData={formatNoteReferrerEntityData(location.state)}
+        noteId={noteId}
+      />
+    )
+    : <Redirect to="/requests" />;
+};
+
+NoteViewRoute.propTypes = {
+  history: ReactRouterPropTypes.history,
```
Comment (VictorSoroka1) [inline src/routes/NoteViewRoute.js:33]: Should all props be required?
```suggestion
  history: ReactRouterPropTypes.history.isRequired,
```

**Pull request merged**